pkg/controller/account/iam.go

```diff
@@ -731,3 +731,207 @@ func (r *ReconcileAccount) createManagedOpenShiftSupportRole(reqLogger logr.Logg
+	// Check if secret exist in Kubernetes
+	secretExists, err := r.DoesSecretExist(kubeSecretNamespacedName)
+	if err != nil {
+		reqLogger.Error(err, fmt.Sprintf("Unable check if secret: %s exists", secretNamespacedName.String()))
+		return err
+	}
+
+	if !secretExists {
+		// If secret doesn't exist, create new one
+		secretName, err = r.BuildIAMUser(reqLogger, awsAssumedRoleClient, currentAcctInstance, iamUserUHC, nameSpace)
+		if err != nil {
+			reason, errType := getBuildIAMUserErrorReason(err)
+			errMsg := fmt.Sprintf("Failed to recreate IAM UHC user %s: %s", iamUserUHC, err)
+			_, stateErr := r.setAccountFailed(
+				reqLogger,
+				currentAcctInstance,
+				errType,
+				reason,
+				errMsg,
+				AccountFailed,
+			)
+			if stateErr != nil {
+				reqLogger.Error(err, "failed setting account state", "desiredState", AccountFailed)
+			}
+			return err
+		}
+	} else {
+		// If secret exists, check that credentials are valid
+		validSecret, err := r.IsKubeSecretValid(reqLogger, currentAcctInstance)
+		if err != nil {
+			reqLogger.Error(err, "failed secret validation")
+		}
+		if !validSecret {
+			// If credentials aren't valid, make them valid again
+			secretName, err = r.RevalidateSecret(reqLogger, awsAssumedRoleClient, currentAcctInstance, iamUserUHC, nameSpace)
+			if err != nil {
+				reason, errType := getBuildIAMUserErrorReason(err)
+				errMsg := fmt.Sprintf("Failed to revalidate IAM UHC user %s: %s", iamUserUHC, err)
+				_, stateErr := r.setAccountFailed(
+					reqLogger,
+					currentAcctInstance,
+					errType,
+					reason,
+					errMsg,
+					AccountFailed,
+				)
+				if stateErr != nil {
+					reqLogger.Error(err, "failed setting account state", "desiredState", AccountFailed)
+				}
+				return err
+			}
+		}
+	}
+
+	currentAcctInstance.Spec.IAMUserSecret = *secretName
+	err = r.accountSpecUpdate(reqLogger, currentAcctInstance)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *ReconcileAccount) IsKubeSecretValid(reqLogger logr.Logger, currentAcctInstance *awsv1alpha1.Account) (bool, error) {
+
+	// Build new aws client with credentials inside secret
+	awsClient, err := r.awsClientBuilder.GetClient(controllerName, r.Client, awsclient.NewAwsClientInput{
+		SecretName: currentAcctInstance.Spec.IAMUserSecret,
+		NameSpace:  currentAcctInstance.Namespace,
+		AwsRegion:  "us-east-1",
+	})
+	if err != nil {
+		reqLogger.Error(err, "Unable to create aws client")
+		return false, err
+	}
+
+	// Make aws call to check if credentials are valid
+	gciOut, err := awsClient.GetCallerIdentity(&sts.GetCallerIdentityInput{})
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case "AccessDenied", "InvalidClientToken":
+				reqLogger.Error(err, "invalid credentials provided")
+				return false, err
+			}
+			reqLogger.Error(err, "failed to get caller identity")
+			return false, err
+		}
+	}
+
+	if gciOut != nil {
```

Comment:
this is if statement is redundant and can just be omitted.

Reply:
the variable gciOut can't be left unused

Reply:
Then you can also just go ahead an not define it in the first place, like:

```
_, err := awsClient.GetCalleridentity(....)
```
Or am I missing something? Cause it seems like `gciOut` isn't really used anywhere.